tests/unit/base/MockObject.h

```diff
@@ -0,0 +1,28 @@
+#include <shogun/base/SGObject.h>
+#include <shogun/lib/SGVector.h>
+
+namespace shogun
+{
+class CMockObject : public CSGObject
```

Comment:
give this guy a mini `@brief` so that other devs know why you create this class here.
I gues there are two points
 * testing registering member variables in tag framework
 * avoiding member variables using tag framework